### PR TITLE
Fixed room and localization schema

### LINKED ISSUE #74: fix room and localization schema (effort 3)
room shouldnt have hotelId
zip code should be string
number should be a function in validation

check imports and exports everywhere,
check eg. if hotelId(etc) is changed to hotel and change wherever it is needed. Schema has been changed

remove Address from models/adrress.js
remove Room from models/room.js

remember to change it in rotues, schemas and other models

test it

### PR DESCRIPTION
**Pull request by GoraJakub**
Moreover I fixed hotel routes (I hope) and add some changes, don't mind Joivaildation yet, it will be changed in #76

### CODE REVIEW

File: models/reservation.js [hunk 1]
```diff
@@ -57,4 +57,4 @@ reservationSchema.methods.toJSON = function () {
 
 const Reservation = mongoose.model('Reservation', reservationSchema)
 
-module.exports = Reservation
+exports.Reservation = Reservation
```
Comment (bartoszbialecki) [inline models/reservation.js]: don't do this, it should be default export

File: models/room.js [hunk 2]
```diff
@@ -27,22 +27,19 @@ const roomSchema = new mongoose.Schema({
   },
 })
 
-const Room = mongoose.model('Room', roomSchema)
-
 const validateRoom = (room) => {
   const schema = Joi.object({
     hotelId: Joi.objectId().required(),
```
Comment (pdybowski) [inline models/room.js]: image


instead add roomNumber like in schema

**Review by pdybowski — COMMENTED** [review]
fix

File: services/ownerService.js [hunk 3]
```diff
@@ -51,7 +51,9 @@ exports.updateHotel = async (id, data) => {
   return hotel
 }
 
-exports.deleteHotel = async (id, isForceDelete) => {
+exports.deleteHotel = async (ownerId, id, isForceDelete) => {
+  const hotel = await Hotel.findById(id)
+  if (hotel.ownerId !== ownerId) throw new ApiError(403, 'Forbidden')
```
Comment (pdybowski) [inline services/ownerService.js:56]: First check if hotel exists. When it is undefined we will get an error: ownerId is not a Property of undefined.. Something  like that

File: controllers/ownerController.js [hunk 4]
```diff
@@ -80,13 +81,14 @@ exports.deleteHotel = async (req, res, next) => {
...
     const isForceDelete = forceDelete === 'true'
-    await deleteHotel(req.params.id, isForceDelete)
+    await deleteHotel(req.user._id, req.params.id, isForceDelete)
     res.sendStatus(200)
   } catch (error) {
     if (error instanceof mongoose.Error.CastError) {
       next(new ApiError(400, 'Hotel with given ID does not exist'))
     }
-    next(new ApiError(400, error))
+    console.log(error.message)
```
Comment (pdybowski) [inline controllers/ownerController.js]: Remove console.log

File: services/ownerService.js (hunk 3 above)
Comment (pdybowski) [inline services/ownerService.js:56]: First check if hotel exists

**Review by pdybowski — COMMENTED** [review]
Fix

File: models/hotel.js [hunk 5]
```diff
@@ -67,7 +67,7 @@ const validateHotel = (hotel) => {
     name: Joi.string().min(1).required(),
     clientsRate: Joi.array().items(schemaRate),
     email: Joi.string().email().required(),
-    description: Joi.string().min(0).required(),
+    description: Joi.string().min(0),
```
Comment (pdybowski) [inline models/hotel.js]: Will min(0) work with a string? Xd

**Review by pdybowski — COMMENTED** [review]
Question

**Pull request merged**